csrc/dynamic_transform.cpp

```diff
@@ -772,7 +777,8 @@ bool DynamicTransformConcretizer::propagateFromProducerToConsumer(
 
       auto input_id = c2p.at(root_id);
       NVF_ERROR(
-          input_id->getIterType() != IterType::Symbolic,
+          input_id == maybeMutated(input_id) &&
```

Comment:
I'd like to see more descriptive error message if this isn't the case. For example, I'd like to see what `maybeMutated(input_id)` would look like. 

Shouldn't this assertion and the one below about itertype separated?

Reply:
Indeed. Done.